tests/reducers/helpers/version-comparison.test.js

```diff
@@ -0,0 +1,23 @@
+import compareVersions from "../../../src/reducers/helpers/version-comparison"
+
+describe('Version Comparison', () => {
+    test('4.0.0 compared to 3.0.1 is 1', () => {
+        expect(compareVersions('4.0.0', '3.0.1')).toBe(1);
+    });
+
+    test('3.0.1 compared to 4.0.0 is -1', () => {
+        expect(compareVersions('3.0.1', '4.0.0')).toBe(-1);
+    });
+
+    test('4.0.0 compared to 3.0.1 is 0', () => {
+        expect(compareVersions('3.0.1', '3.0.1')).toBe(0);
+    })
+
+    test('3.0.1 compared to 1.0.2 is 1', () => {
+        expect(compareVersions('3.0.1', '1.0.2')).toBe(1);
+    })
+
+    test('3.0.1 compared to 3.0.0 is 1', () => {
```

Comment:
Consider a test case where one version has fewer components than the other (to test the "right pad with zeros" logic).